projects/vdk-core/src/vdk/internal/builtin_plugins/termination_message/writer.py

```diff
@@ -29,9 +29,10 @@ def vdk_configure(self, config_builder: ConfigurationBuilder):
 
     @hookimpl
     def vdk_exit(self, context: CoreContext, exit_code: int):
+        not_resolved_predicate = lambda e: not e.resolved
         self.write_termination_message(
-            errors.get_blamee_overall(),  # TODO: get this from context
-            errors.get_blamee_overall_user_error(),  # TODO: get this from context
+            errors.get_blamee_overall(not_resolved_predicate),
+            errors.get_blamee_overall_user_error(not_resolved_predicate),
```

Comment:
The purpose of those TODOs is to figure out a solution which doesn't require keeping global state - which I think it's very problematic. We can probably refactor it in a way that the error state is kept in the JobContext and that would simplify the codebase. 

This change doesn't address this problem though so I'd keep the TODOs.

Reply:
@tozka I've described in the PR summary, that I went for resolving those todos, in the scope of this PR. 

So the todos suggest to make errors registry inner non-static of *Context instead), and the proposal in the scope of this PR is to keep the errors context static. The reason is, that errors registry needs to be ready and available at earlier stages of the VDK life-cycle (than any context initialisation) - due to errors could occur earlier, so we need to handle those.

Looking at the `core_hook_spec.py` API, most of the hooks are actually happening before CoreContext and especially JobContext initialization. I think the todos are wrong to move, and my proposal is to cleanup those.

What do you think, do you have anything else in mind?

Reply:
> The reason is, that errors registry needs to be ready and available at earlier stages of the VDK life-cycle (than any context initialisation) - due to errors could occur earlier, so we need to handle those.

That makes a lot of sense. Ok. Since I need to shift my thinking I will take a new look at the review. I will do that tomorrow though. Sorry I didn't notice the comment earlier today.

Reply:
@tozka ping